src/main/java/com/commercetools/sync/products/utils/ProductSyncUtils.java

```diff
@@ -104,7 +106,24 @@
         // lastly publish/unpublish product
         buildPublishUpdateAction(oldProduct, newProduct).ifPresent(updateActions::add);
 
-        return updateActions;
+        return prioritizeUpdateActions(updateActions);
+    }
+
+    private static List<UpdateAction<Product>> prioritizeUpdateActions(
```

Comment:
No, it doesn't make any problems.
I tried it my self and e2e is working fine in this regard.
Basically making the "removeVariant" on top of the list of the update actions will cause some issues when it comes to changing the master variant on the target product.
For instance, changing the master variant would involve the following set of actions:
"addVariant"
"changeMasterVariant"
"removeVariant"

so, in this case making "removeVariant" will cause an error saying that you can't remove the variant because it's a master variant.

So, i saw that the most important action to be on top of the list of update actions is the "setAttributeInAllVariants" because you want to keep the target product consistent first before applying "addVariant" actions for example and it doesn't cause any problems whe it comes to "removeVariant".

Reply:
I totally agree to this, I just wanted to keep it simple for now but i can make the changes accordingly.